### ISSUE TITLE
[Remote-SSH Bug]: go up folder button stop working randomly

### ISSUE THREAD

**Issue by xgdgsc** [microsoft/vscode-remote-release#9797]
### Is there an existing issue for this bug?

- [X] I have searched the existing issues

### Required Troubleshooting Steps

- [X] I have followed these troubleshooting steps
- [ ] I have tried both values of the `remote.SSH.useLocalServer` setting

### Connect Locally

It connects successfully

### Expected Behavior

Remote folder selection menu works properly with go up folder button.

### Steps To Reproduce

open remote and use File->Open File/Folder . The go up folder button stops working randomly as the gif shows:
![动画](https://github.com/microsoft/vscode-remote-release/assets/1189869/2a74f3ff-d598-4f6c-b941-98c281d345e6)

### Remote-SSH Log

<details>
<summary>Remote-SSH Log</summary>
<p>

```
[09:01:40.017] Log Level: 2
[09:01:40.087] SSH Resolver called for "ssh-remote+r99", attempt 1
[09:01:40.088] "remote.SSH.useLocalServer": false
[09:01:40.089] "remote.SSH.useExecServer": true
[09:01:40.093] "remote.SSH.showLoginTerminal": false
[09:01:40.093] "remote.SSH.remotePlatform": {"hyremote-sgao":"linux","hosted-lox":"linux","hyremote-jump":"linux","tencent-cloud118":"linux","hyremote-cas1-sgao":"linux","hyremote-r2-data_oper":"linux","hyremote-r6-sgao":"linux","hosted-hyremote-jump":"linux","hosted-backup":"linux","hyremote-r2-trd_oper":"linux","r5-data_oper":"linux","hyremote-d1-data_oper":"linux","r5-sgao":"linux","gctime.space":"linux","r99":"linux","r100":"linux"}
[09:01:40.093] "remote.SSH.path": C:\Windows\System32\OpenSSH\ssh.exe
[09:01:40.094] "remote.SSH.configFile": undefined
[09:01:40.094] "remote.SSH.useFlock": true
[09:01:40.094] "remote.SSH.lockfilesInTmp": false
[09:01:40.094] "remote.SSH.localServerDownload": auto
[09:01:40.095] "remote.SSH.remoteServerListenOnSocket": false
[09:01:40.095] "remote.SSH.showLoginTerminal": false
[09:01:40.095] "remote.SSH.defaultExtensions": []
[09:01:40.095] "remote.SSH.loglevel": 2
[09:01:40.095] "remote.SSH.enableDynamicForwarding": true
[09:01:40.095] "remote.SSH.enableRemoteCommand": false
[09:01:40.096] "remote.SSH.serverPickPortsFromRange": {}
[09:01:40.096] "remote.SSH.serverInstallPath": {}
[09:01:40.100] VS Code version: 1.88.1
[09:01:40.100] Remote-SSH version: remote-ssh@0.110.1
[09:01:40.100] win32 arm64
[09:01:40.108] SSH Resolver called for host: r99
[09:01:40.108] Setting up SSH remote "r99"
[09:01:40.114] Using commit id "e170252f762678dec6ca2cc69aba1570769a5d39" and quality "stable" for server
[09:01:40.119] Install and start server if needed
[09:01:40.159] Checking ssh with "C:\Windows\System32\OpenSSH\ssh.exe -V"
[09:01:43.397] > OpenSSH_for_Windows_8.6p1, LibreSSL 3.4.3

[09:01:43.424] Running script with connection command: "C:\Windows\System32\OpenSSH\ssh.exe" -T -D 23982 r99 bash
[09:01:43.440] Terminal shell path: C:\WINDOWS\System32\cmd.exe
[09:01:47.481] > /home/user1/.bashrc: 第 10 行:bind: 警告:行编辑未启用
> fee4b021bb04: running
> /home/user1/.bashrc: 第 11 行:bind: 警告:行编辑未启用
> ]0;C:\WINDOWS\System32\cmd.exe
[09:01:47.482] Got some output, clearing connection timeout
[09:01:47.501] > Found existing installation at /home/user1/.vscode-server...
> Starting VS Code CLI... "/home/user1/.vscode-server/code-e170252f762678dec6ca2cc69aba1570769a5d39" command-shell --cli-data-dir "/home/user1/.vscode-server/cli" --on-port --parent-process-id 75816 &> "/home/user1/.vscode-server/.cli.e170252f762678dec6ca2cc69aba1570769a5d39.log" < /dev/null
> printenv:
>     AUTOJUMP_ERROR_PATH=/home/user1/.local/share/autojump/errors.log
>     XDG_SESSION_ID=70231
>     AUTOJUMP_SOURCED=1
>     SHELL=/bin/bash
>     SSH_CLIENT=192.168.101.178 54750 23982
>     USER=user1
>     JULIAUP_SERVER=https://mirrors.sjtug.sjtu.edu.cn/julia-releases/
>     JULIA_PKG_SERVER=https://mirrors.cernet.edu.cn/julia
>     VSCODE_AGENT_FOLDER=/home/user1/.vscode-server
>     PATH=/n1/env_centos/7.6/cmake/cmake-3.20.0-linux-x86_64/bin/:/home/user1/.juliaup/bin:/usr/local/bin:/usr/bin
>     MAIL=/var/mail/user1
>     PWD=/home/user1
>     LANG=zh_CN.UTF-8
>     HOME=/home/user1
>     SHLVL=2
>     LOGNAME=user1
>     SSH_CONNECTION=192.168.101.178 54750 192.168.101.169 23982
>     LESSOPEN=||/usr/bin/lesspipe.sh %s
>     XDG_RUNTIME_DIR=/run/user/1001
>     _=/usr/bin/printenv
> Removing old logfile at /home/user1/.vscode-server/.cli.e170252f762678dec6ca2cc6
> 69aba1570769a5d39.log
> Spawned remote CLI: 75884
[09:01:47.528] > 
[09:01:47.546] > Waiting for server log...
[09:01:47.579] > 
[09:01:47.647] > fee4b021bb04: start
> SSH_AUTH_SOCK====
> DISPLAY====
> listeningOn==127.0.0.1:44894==
> osReleaseId==centos==
> arch==x86_64==
> vscodeArch==x64==
> bitness==64==
> tmpDir==/run/user/1001==
> platform==linux==
> unpackResult====
> didLocalDownload==0==
> downloadTime====
> installTime====
> serverStartTime==102==
> execServerToken==1aaa1111-a111-1a11-aa11-a11a111aa11a==
> fee4b021bb04: end
[09:01:47.648] Received install output: 
SSH_AUTH_SOCK====
DISPLAY====
listeningOn==127.0.0.1:44894==
osReleaseId==centos==
arch==x86_64==
vscodeArch==x64==
bitness==64==
tmpDir==/run/user/1001==
platform==linux==
unpackResult====
didLocalDownload==0==
downloadTime====
installTime====
serverStartTime==102==
execServerToken==1aaa1111-a111-1a11-aa11-a11a111aa11a==

[09:01:47.649] Remote server is listening on port 44894
[09:01:47.649] Parsed server configuration: {"serverConfiguration":{"remoteListeningOn":{"port":44894},"osReleaseId":"centos","arch":"x86_64","sshAuthSock":"","display":"","tmpDir":"/run/user/1001","platform":"linux","execServerToken":"1aaa1111-a111-1a11-aa11-a11a111aa11a"},"serverStartTime":102,"installUnpackCode":""}
[09:01:47.663] Starting forwarding server. local port 24007 -> socksPort 23982 -> remotePort 44894
[09:01:47.664] Forwarding server listening on port 24007
[09:01:47.664] Waiting for ssh tunnel to be ready
[09:01:47.665] > 
[09:01:47.666] [Forwarding server port 24007] Got connection 0
[09:01:47.668] Tunneled port 44894 to local port 24007
[09:01:47.668] Resolved "ssh-remote+r99" to "port 24007"
[09:01:47.702] Initizing new exec server for ssh-remote+r99
[09:01:47.703] Resolving exec server at port 24007
[09:01:47.781] [Forwarding server port 24007] Got connection 1
[09:01:47.882] Exec server for ssh-remote+r99 created and cached
[09:01:47.900] ------




[09:01:47.934] [server] Checking /home/user1/.vscode-server/cli/servers/Stable-e170252f762678dec6ca2cc69aba1570769a5d39/log.txt and /home/user1/.vscode-server/cli/servers/Stable-e170252f762678dec6ca2cc69aba1570769a5d39/pid.txt for a running server...
[09:01:48.008] [server] Installing and setting up Visual Studio Code Server...
[09:01:48.009] [server] Server setup complete
[09:01:48.009] [server] Starting server...
[09:01:48.010] [server] Starting server with command... Command { std: "/home/user1/.vscode-server/cli/servers/Stable-e170252f762678dec6ca2cc69aba1570769a5d39/server/bin/code-server" "--connection-token=remotessh" "--accept-server-license-terms" "--start-server" "--enable-remote-auto-shutdown" "--socket-path=/tmp/code-8a9f58ac-56e8-4cf9-b653-d5b4c7597065", kill_on_drop: false }
[09:01:48.010] [server] *
[09:01:48.052] [server] * Visual Studio Code Server
[09:01:48.052] [server] *
[09:01:48.053] [server] * By using the software, you agree to
[09:01:48.053] [server] * the Visual Studio Code Server License Terms (https://aka.ms/vscode-server-license) and
[09:01:48.053] [server] * the Microsoft Privacy Statement (https://privacy.microsoft.com/en-US/privacystatement).
[09:01:48.053] [server] *
[09:01:48.053] [server] Server bound to /tmp/code-8a9f58ac-56e8-4cf9-b653-d5b4c7597065
[09:01:48.053] [server] Extension host agent listening on /tmp/code-8a9f58ac-56e8-4cf9-b653-d5b4c7597065
[09:01:48.054] [server] parsed location: "/tmp/code-8a9f58ac-56e8-4cf9-b653-d5b4c7597065"
[09:01:48.054] [server] 
[09:01:48.061] [server] Server started
[09:01:48.126] [server] [09:01:46] 
[09:01:48.126] [server] 
[09:01:48.127] [server] 
[09:01:48.127] [server] 
[09:01:48.127] [server] 
[09:01:48.269] [server] [09:01:46] Extension host agent started.
[09:01:48.270] [server] [09:01:46] [cmake-tools]: Couldn't find message for key cmake-tools.configuration.cmake.options.advanced.statusBarLength.description.
[09:01:48.270] [server] [09:01:46] [cmake-tools]: Couldn't find message for key cmake-tools.configuration.cmake.options.advanced.statusBarLength.description.
[09:01:48.277] [server] [09:01:46] [cmake-tools]: Couldn't find message for key cmake-tools.configuration.cmake.options.advanced.statusBarLength.description.
[09:01:48.277] [server] [09:01:46] [cmake-tools]: Couldn't find message for key cmake-tools.configuration.cmake.options.advanced.statusBarLength.description.
[09:01:48.277] [server] [09:01:46] [cmake-tools]: Couldn't find message for key cmake-tools.configuration.cmake.options.advanced.statusBarLength.description.
[09:01:48.277] [server] [09:01:46] [cmake-tools]: Couldn't find message for key cmake-tools.configuration.cmake.options.advanced.statusBarLength.description.
[09:01:48.278] [server] [09:01:46] [cmake-tools]: Couldn't find message for key cmake-tools.configuration.cmake.options.advanced.statusBarLength.description.
[09:01:48.279] [server] [09:01:46] [cmake-tools]: Couldn't find message for key cmake-tools.configuration.cmake.options.advanced.statusBarLength.description.
[09:01:48.279] [server] [09:01:46] [cmake-tools]: Couldn't find message for key cmake-tools.configuration.cmake.options.advanced.statusBarLength.description.
[09:01:48.279] [server] [09:01:46] [cmake-tools]: Couldn't find message for key cmake-tools.configuration.cmake.options.advanced.statusBarLength.description.
[09:01:48.279] [server] [09:01:46] [cmake-tools]: Couldn't find message for key cmake-tools.configuration.cmake.options.advanced.statusBarLength.description.
[09:01:48.279] [server] [09:01:46] [cmake-tools]: Couldn't find message for key cmake-tools.configuration.cmake.options.advanced.statusBarLength.description.
[09:01:48.287] [server] [09:01:46] [cmake-tools]: Couldn't find message for key cmake-tools.configuration.cmake.options.advanced.statusBarLength.description.
[09:01:48.289] [server] [09:01:46] [cmake-tools]: Couldn't find message for key cmake-tools.configuration.cmake.options.advanced.statusBarLength.description.
[09:01:48.289] [server] [09:01:46] [cmake-tools]: Couldn't find message for key cmake-tools.configuration.cmake.options.advanced.statusBarLength.description.
[09:01:48.290] [server] [09:01:46] [cmake-tools]: Couldn't find message for key cmake-tools.configuration.cmake.options.advanced.statusBarLength.description.
[09:01:48.290] [server] [09:01:46] [cmake-tools]: Couldn't find message for key cmake-tools.configuration.cmake.options.advanced.statusBarLength.description.
[09:01:48.290] [server] [09:01:46] [cmake-tools]: Couldn't find message for key cmake-tools.configuration.cmake.options.advanced.statusBarLength.description.
[09:01:48.291] [server] [09:01:46] [cmake-tools]: Couldn't find message for key cmake-tools.configuration.cmake.options.advanced.statusBarLength.description.
[09:01:48.291] [server] [09:01:46] [cmake-tools]: Couldn't find message for key cmake-tools.configuration.cmake.options.advanced.statusBarLength.description.
[09:01:48.291] [server] [09:01:46] [cmake-tools]: Couldn't find message for key cmake-tools.configuration.cmake.options.advanced.statusBarLength.description.
[09:01:48.291] [server] [09:01:46] [cmake-tools]: Couldn't find message for key cmake-tools.configuration.cmake.options.advanced.statusBarLength.description.
[09:01:48.293] [server] [09:01:46] [cmake-tools]: Couldn't find message for key cmake-tools.configuration.cmake.options.advanced.statusBarLength.description.
[09:01:48.293] [server] [09:01:46] [cmake-tools]: Couldn't find message for key cmake-tools.configuration.cmake.options.advanced.statusBarLength.description.
[09:01:48.639] [server] [09:01:46] [<unknown>][935275f4][ManagementConnection] New connection established.
[09:01:48.766] [server] [09:01:46] [<unknown>][3bef1f35][ExtensionHostConnection] New connection established.
[09:01:48.784] [server] [09:01:46] [<unknown>][3bef1f35][ExtensionHostConnection] <76165> Launched Extension Host Process.
[09:01:49.164] [server] [09:01:47] [cmake-tools]: Couldn't find message for key cmake-tools.configuration.cmake.options.advanced.statusBarLength.description.
[09:01:49.164] [server] [09:01:47] [cmake-tools]: Couldn't find message for key cmake-tools.configuration.cmake.options.advanced.statusBarLength.description.
[09:01:49.164] [server] [09:01:47] [cmake-tools]: Couldn't find message for key cmake-tools.configuration.cmake.options.advanced.statusBarLength.description.
[09:01:49.164] [server] [09:01:47] [cmake-tools]: Couldn't find message for key cmake-tools.configuration.cmake.options.advanced.statusBarLength.description.
[09:01:49.165] [server] [09:01:47] [cmake-tools]: Couldn't find message for key cmake-tools.configuration.cmake.options.advanced.statusBarLength.description.
[09:01:49.165] [server] [09:01:47] [cmake-tools]: Couldn't find message for key cmake-tools.configuration.cmake.options.advanced.statusBarLength.description.
[09:01:49.165] [server] [09:01:47] [cmake-tools]: Couldn't find message for key cmake-tools.configuration.cmake.options.advanced.statusBarLength.description.
[09:01:49.165] [server] [09:01:47] [cmake-tools]: Couldn't find message for key cmake-tools.configuration.cmake.options.advanced.statusBarLength.description.
[09:01:49.165] [server] [09:01:47] [cmake-tools]: Couldn't find message for key cmake-tools.configuration.cmake.options.advanced.statusBarLength.description.
[09:01:49.166] [server] [09:01:47] [cmake-tools]: Couldn't find message for key cmake-tools.configuration.cmake.options.advanced.statusBarLength.description.
[09:01:49.166] [server] [09:01:47] [cmake-tools]: Couldn't find message for key cmake-tools.configuration.cmake.options.advanced.statusBarLength.description.
[09:01:49.166] [server] [09:01:47] [cmake-tools]: Couldn't find message for key cmake-tools.configuration.cmake.options.advanced.statusBarLength.description.
[09:01:49.219] [server] [09:01:47] [cmake-tools]: Couldn't find message for key cmake-tools.configuration.cmake.options.advanced.statusBarLength.description.
[09:01:49.220] [server] [09:01:47] [cmake-tools]: Couldn't find message for key cmake-tools.configuration.cmake.options.advanced.statusBarLength.description.
[09:01:49.220] [server] [09:01:47] [cmake-tools]: Couldn't find message for key cmake-tools.configuration.cmake.options.advanced.statusBarLength.description.
[09:01:49.220] [server] [09:01:47] [cmake-tools]: Couldn't find message for key cmake-tools.configuration.cmake.options.advanced.statusBarLength.description.
[09:01:49.220] [server] [09:01:47] [cmake-tools]: Couldn't find message for key cmake-tools.configuration.cmake.options.advanced.statusBarLength.description.
[09:01:49.221] [server] [09:01:47] [cmake-tools]: Couldn't find message for key cmake-tools.configuration.cmake.options.advanced.statusBarLength.description.
[09:01:49.221] [server] [09:01:47] [cmake-tools]: Couldn't find message for key cmake-tools.configuration.cmake.options.advanced.statusBarLength.description.
[09:01:49.221] [server] [09:01:47] [cmake-tools]: Couldn't find message for key cmake-tools.configuration.cmake.options.advanced.statusBarLength.description.
[09:01:49.236] [server] [09:01:47] [cmake-tools]: Couldn't find message for key cmake-tools.configuration.cmake.options.advanced.statusBarLength.description.
[09:01:49.238] [server] [09:01:47] [cmake-tools]: Couldn't find message for key cmake-tools.configuration.cmake.options.advanced.statusBarLength.description.
[09:01:49.238] [server] [09:01:47] [cmake-tools]: Couldn't find message for key cmake-tools.configuration.cmake.options.advanced.statusBarLength.description.
[09:01:49.246] [server] [09:01:47] [cmake-tools]: Couldn't find message for key cmake-tools.configuration.cmake.options.advanced.statusBarLength.description.
[09:01:49.524] [server] [09:01:47] [cmake-tools]: Couldn't find message for key cmake-tools.configuration.cmake.options.advanced.statusBarLength.description.
[09:01:49.526] [server] [09:01:47] [cmake-tools]: Couldn't find message for key cmake-tools.configuration.cmake.options.advanced.statusBarLength.description.
[09:01:49.526] [server] [09:01:47] [cmake-tools]: Couldn't find message for key cmake-tools.configuration.cmake.options.advanced.statusBarLength.description.
[09:01:49.527] [server] [09:01:47] [cmake-tools]: Couldn't find message for key cmake-tools.configuration.cmake.options.advanced.statusBarLength.description.
[09:01:49.527] [server] [09:01:47] [cmake-tools]: Couldn't find message for key cmake-tools.configuration.cmake.options.advanced.statusBarLength.description.
[09:01:49.527] [server] [09:01:47] [cmake-tools]: Couldn't find message for key cmake-tools.configuration.cmake.options.advanced.statusBarLength.description.
[09:01:49.527] [server] [09:01:47] [cmake-tools]: Couldn't find message for key cmake-tools.configuration.cmake.options.advanced.statusBarLength.description.
[09:01:49.528] [server] [09:01:47] [cmake-tools]: Couldn't find message for key cmake-tools.configuration.cmake.options.advanced.statusBarLength.description.
[09:01:49.528] [server] [09:01:47] [cmake-tools]: Couldn't find message for key cmake-tools.configuration.cmake.options.advanced.statusBarLength.description.
[09:01:49.529] [server] [09:01:47] [cmake-tools]: Couldn't find message for key cmake-tools.configuration.cmake.options.advanced.statusBarLength.description.
[09:01:49.529] [server] [09:01:47] [cmake-tools]: Couldn't find message for key cmake-tools.configuration.cmake.options.advanced.statusBarLength.description.
[09:01:49.529] [server] [09:01:47] [cmake-tools]: Couldn't find message for key cmake-tools.configuration.cmake.options.advanced.statusBarLength.description.
[09:01:53.581] [server] [09:01:50] [cmake-tools]: Couldn't find message for key cmake-tools.configuration.cmake.options.advanced.statusBarLength.description.
[09:01:53.587] [server] [09:01:50] [cmake-tools]: Couldn't find message for key cmake-tools.configuration.cmake.options.advanced.statusBarLength.description.
[09:01:53.590] [server] [09:01:50] [cmake-tools]: Couldn't find message for key cmake-tools.configuration.cmake.options.advanced.statusBarLength.description.
[09:01:53.590] [server] [09:01:50] [cmake-tools]: Couldn't find message for key cmake-tools.configuration.cmake.options.advanced.statusBarLength.description.
[09:01:53.591] [server] [09:01:50] [cmake-tools]: Couldn't find message for key cmake-tools.configuration.cmake.options.advanced.statusBarLength.description.
[09:01:53.591] [server] [09:01:50] [cmake-tools]: Couldn't find message for key cmake-tools.configuration.cmake.options.advanced.statusBarLength.description.
[09:01:53.591] [server] [09:01:50] [cmake-tools]: Couldn't find message for key cmake-tools.configuration.cmake.options.advanced.statusBarLength.description.
[09:01:53.591] [server] [09:01:50] [cmake-tools]: Couldn't find message for key cmake-tools.configuration.cmake.options.advanced.statusBarLength.description.
[09:01:53.592] [server] [09:01:50] [cmake-tools]: Couldn't find message for key cmake-tools.configuration.cmake.options.advanced.statusBarLength.description.
[09:01:53.593] [server] [09:01:50] [cmake-tools]: Couldn't find message for key cmake-tools.configuration.cmake.options.advanced.statusBarLength.description.

 

```

</p>
</details>

**Comment by ivaylo-madjarov** [comment]
I see this in WSL also, in addition to SSH, so it might be a more general VS Code issue. It allows you to go up a folder one time, but not twice. 

Furthermore, if you have two folders that are named the same thing, you can't click through the second one. For example, if you start in /home/user/, then go to /home/user/folder/, then try to click through to /home/user/folder/folder/, it also fails. 

I am on VS Code 1.88.1.

**Comment by xgdgsc** [comment]
Fixed in 1.90
